### ISSUE TITLE
Handling symlinks on Windows

### ISSUE THREAD

**Issue by ojeytonwilliams** [freeCodeCamp/chapter#465]
@paizle 's issue seems to be related to Git Bash + Windows and probably because there's a symlink involved.

![image](https://user-images.githubusercontent.com/1777776/112842534-a4778d00-906f-11eb-9b81-d63f25b299ae.png)

@ojeytonwilliams do we think using the literal path instead of a symlink is likely to fix one environment and break others? If so, perhaps we break that into a new issue.

I found some chat about the [escaping of characters being an issue / bug with Git Bash on Windows](https://github.com/freeCodeCamp/chapter/issues/447#issuecomment-794076923) so symlink and/or escaping seems like they could the what's breaking Git Bash.

_Originally posted by @allella in https://github.com/freeCodeCamp/chapter/issues/461#issuecomment-809370861_

**Comment by paizle** [comment]
I think enabling symlinks did the trick. I just had to re-install Git Bash with the symlink option checked.

This works fine now:
`"typeorm": "ts-node -P tsconfig.server.json ./node_modules/.bin/typeorm",`

Subsequently, I tried the following but node complained about the single quotes:
`"both": "concurrently 'npm run dev' 'cd client && npm run dev'",`

Relevant error:
`npm ERR! chapter@0.0.1 both: `concurrently 'npm run dev' 'cd client && npm run d
ev'`
`

**Comment by ojeytonwilliams** [comment]
Okay, awesome, thanks @paizle.  Would you mind creating a PR with a brief addition to the docs?  Since you've just done this, I think you're best placed to explain the required steps.

Regarding the single quotes, I've kept your suggestion in https://github.com/freeCodeCamp/chapter/pull/461/, so that will get resolved.

**Comment by allella** [comment]
@paizle  I can add something to the "Running the Application" Step 1 with a note and link about Git Bash requiring the symlink option.

Sound good?

**Comment by allella** [comment]
@paizle it looks like this can be [enabled on a per clone basis](https://github.com/git-for-windows/git/wiki/Symbolic-Links#background) instead of turned on by default.

`git clone -c core.symlinks=true <URL>
`

There are some a list or reasons it's not on by default on that page.

Could you re-install Git Bash with the symlinks turned off and then try a fresh clone of the **Chapter** repository using the `-c core.symlinks=true` option shown above, and let us know if that works?

**Comment by paizle** [comment]
> 
> 
> Okay, awesome, thanks @paizle. Would you mind creating a PR with a brief addition to the docs? Since you've just done this, I think you're best placed to explain the required steps.
> 
> Regarding the single quotes, I've kept your suggestion in #461, so that will get resolved.

Yep, will do.

**Comment by paizle** [comment]
> 
> 
> @paizle it looks like Git Bash symlinks can be [enabled on a per clone basis](https://github.com/git-for-windows/git/wiki/Symbolic-Links#background) instead of turned on by default.
> 
> `git clone -c core.symlinks=true <URL> `
> 
> There are some reasons listed as to why Git Bash symlinks are not enabled by default on that page.
> 
> Could you re-install Git Bash with the symlinks turned off and then try a fresh clone of the **Chapter** repository using the `-c core.symlinks=true` option shown above, and let us know if that works?

Ok, re-installed Git Bash without the symlinks and I gave this a try both with and without the core.symlinks config option and both worked.

I tried rebooting and different install options for Git Bash just trying to break it again but alas I could not.

I did have symlinks feature turned off before so I'm not sure what the problem actually was. Possibly it was the newer version of Git Bash that fixed my issues. I'm not sure. 

Seems fine now.

**Comment by allella** [comment]
@paizle I have a fresh Windows laptop I just setup for testing things.  I'll give it a try.

At least now we know with some certainty there are symlink, Windows permissions, and character escaping issues with Git for Windows / Git Bash / Github Desktop (built on top of on the former).

It would be good to figure out how to replicate the errors consistently so we know whatever goes in the docs isn't still going to break others.

**Comment by allella** [comment]
@paizle one other thing that [controls the symlinks is the global config file](https://github.community/t/git-bash-symbolic-links-on-windows/522/7).

I suspect checking the box during the install saves to _C:\ProgramData\Git\config_ so that's another variable we could look at to figure out what's going on.

**Comment by paizle** [comment]
> 
> 
> @paizle one other thing that [controls the symlinks is the global config file](https://github.community/t/git-bash-symbolic-links-on-windows/522/7).
> 
> I suspect checking the box during the install saves to _C:\ProgramData\Git\config_ so that's another variable we could look at to figure out what's going on.

Testing on a fresh windows laptop sounds like a good idea at this point.

**Comment by allella** [comment]
I installed Git for Windows v2.31.1 with the defaults, with symlinks unchecked and confirmed the _core.symlinks=false_  in /etc/gitconfig (also C:\ProgramData\Git\etc\gitconfig) configuration file(s).

I started to install Docker and still waiting on hours of Windows updates before I can install Docker.

#466 includes my notes from running through this as a Windows users.

I'll addressed the symlink / Git Bash part of this issue once I get Docker installed and see if get any issues with Git Bash, or not.
